qiskit/circuit/quantumcircuit.py

```diff
@@ -106,113 +106,859 @@
+    circuit is equal to the :meth:`width` of the circuit::
 
-        Construct a 5-qubit GHZ circuit.
+       assert qc.width() == 12
 
-        .. code-block::
+    We can also just get the number of qubits directly using :attr:`num_qubits`::
 
-           from qiskit import QuantumCircuit
+       assert qc.num_qubits == 12
 
-           qc = QuantumCircuit(5)
-           qc.h(0)
-           qc.cx(0, range(1, 5))
-           qc.measure_all()
+    .. important::
 
-        Construct a 4-qubit Bernstein-Vazirani circuit using registers.
+       For a quantum circuit composed from just qubits, the circuit width is equal
+       to the number of qubits. This is the definition used in quantum computing. However,
+       for more complicated circuits with classical registers, and classically controlled gates,
+       this equivalence breaks down. As such, from now on we will not refer to the number of
+       qubits in a quantum circuit as the width.
 
-        .. plot::
-           :include-source:
+    It is also straightforward to get the number and type of the gates in a circuit using
+    :meth:`count_ops`::
 
-           from qiskit import QuantumRegister, ClassicalRegister, QuantumCircuit
+       qc.count_ops()
 
-           qr = QuantumRegister(3, 'q')
-           anc = QuantumRegister(1, 'ancilla')
-           cr = ClassicalRegister(3, 'c')
-           qc = QuantumCircuit(qr, anc, cr)
+    .. parsed-literal::
 
-           qc.x(anc[0])
-           qc.h(anc[0])
-           qc.h(qr[0:3])
-           qc.cx(qr[0:3], anc[0])
-           qc.h(qr[0:3])
-           qc.barrier(qr)
-           qc.measure(qr, cr)
+       OrderedDict([('cx', 8), ('h', 5), ('x', 3), ('swap', 3)])
 
-           qc.draw('mpl')
+    We can also get just the raw count of operations by computing the circuits
+    :meth:`size`::
+
+       assert qc.size() == 19
+
+    A particularly important circuit property is known as the circuit :meth:`depth`.  The depth
+    of a quantum circuit is a measure of how many "layers" of quantum gates, executed in
+    parallel, it takes to complete the computation defined by the circuit.  Because quantum
+    gates take time to implement, the depth of a circuit roughly corresponds to the amount of
+    time it takes the quantum computer to execute the circuit.  Thus, the depth of a circuit
+    is one important quantity used to measure if a quantum circuit can be run on a device.
+
+    The depth of a quantum circuit has a mathematical definition as the longest path in a
+    directed acyclic graph (DAG).  However, such a definition is a bit hard to grasp, even for
+    experts.  Fortunately, the depth of a circuit can be easily understood by anyone familiar
+    with playing `Tetris <https://en.wikipedia.org/wiki/Tetris>`_.  Lets see how to compute this
+    graphically:
+
+    .. image:: /source_images/depth.gif
+
+    We can verify our graphical result using :meth:`QuantumCircuit.depth`::
+
+       assert qc.depth() == 9
```

Comment:
I realize this is mostly copied over from before, but I wonder if we want to show an example of 2q gate depth and how to compute that using `depth()`. Either here or in the method docstring.

Reply:
I did copy this (near) verbatim.  I think the text of the section is meant to be read as a whole thing, including the image - the `qc` circuit created up top is the same one as shown in the gif, and it does have a depth of 9.

Reply:
I was thinking more back to some offline conversations we were having with about having a documented way to do 2 qubit depth. But I think this probably belongs as an example in the `depth` method's docstring on how to use the filter arg.

Reply:
Done in 32a766964d.